datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -413,13 +415,42 @@ def save_node(
     session.refresh(node.current)
 
 
-def _update_node(
+def update_any_node(
     name: str,
     data: UpdateNode,
     session: Session,
     query_service_client: QueryServiceClient = None,
     current_user: Optional[User] = None,
-):
+) -> Node:
+    """
+    Node update helper function that handles updating any node
+    """
+    node = get_node_by_name(session, name, for_update=True, include_inactive=True)
+    if node.type == NodeType.CUBE:
+        node_revision = update_cube_node(
+            session,
+            node.current,
+            data,
+            query_service_client,
+            current_user,
+        )
+        return node_revision.node if node_revision else node
+    return update_node_with_query(
```

Comment:
Is SOURCE node part of this code path? If so `update_node_with_query` seems to be doing more than its name says. And of not should we catch the SOURCE node case here somehow?